JetBrains DataSpell support on Windows

tsvetilian-ty opened the pull request:
<!--
What GitHub Desktop issue does this PR address? (for example, #1234)
-->

Closes #16020

## Description
- Support for JetBrains DataSpell on Windows

tsvetilian-ty committed:
Add support for JetBrains DataSpell on Windows

tsvetilian-ty committed:
Update supported editors list

tidy-dev commented on docs/technical/editor-integration.md:
```diff
@@ -272,6 +273,7 @@ These editors are currently supported:
  - [Emacs](https://www.gnu.org/software/emacs/)
  - [Lite XL](https://lite-xl.com/)
  - [JetBrains Fleet](https://www.jetbrains.com/fleet/)
+ - [JetBrains DataSpell](https://www.jetbrains.com/dataspell/)
```
```suggestion
```

This is not supported on macOS yet.

tidy-dev reviewed (changes requested):
@tsvetilian-ty Thanks for the integration PR!

I tested on windows and it worked great!

I wasn't going to test on macOS since the PR title indicates this is just for Windows. However, then saw the add to the macOS list in the readme and reviewed macOs integrations and didn't see it. Also tested and it is not added. 

Thus, please either remove the macOS list add or add macOS integration. :)

tsvetilian-ty commented:
@tidy-dev my bad I had to link to the macOS implementation and give more information in my PR. There is a PR with macOS implementation: https://github.com/desktop/desktop/pull/16021, the PR doesn't contain an update to the list, so when I added the Windows integration, I also updated the list for both platforms.

Maybe you could check https://github.com/desktop/desktop/pull/16021 firsts and then merge my PR?

tidy-dev commented:
Ah! I didn't see that. Thank you for pointing it out, I will test it and leave this one be. :)

tidy-dev reviewed (approved):
Looks great! Thanks for pointing out the other PR - Tested and merged so this ready. :)
